### PR TITLE
Fixed various issues in the old Reflectometry GUI

### LINKED ISSUE #18185: Fix some issues in Reflectometry GUI
This refers to the old reflectometry GUI. Issues were reported by Max:

- Processing creates a <runno>_IvsQ workspace which is re-binned according to the setting, but we'd like to have the un-binned IvsQ workspace too and the re-binned one should be called <runno>_IvsQ_binned, like it used to.
- "Transfer" button doesn't work.
- "Save Workspaces" dialog doesn't populate the path correctly, it's missing a slash, e.g.: U:Skoda

### PR DESCRIPTION
**Pull request by PranavBahuguna**
As stated in the relevant issue, this PR fixes the following issues in the old Reflectometry GUI:
- Processing creates a _IvsQ workspace which is re-binned according to the setting, but we'd like to have the un-binned IvsQ workspace too and the re-binned one should be called _IvsQ_binned, like it used to.
- "Transfer" button doesn't work.
- "Save Workspaces" dialog doesn't populate the path correctly, it's missing a slash, e.g.: U:Skoda

**To test:**
- The GUI can be accessed from Interfaces->Reflectometry->ISIS Reflectometry
- Load runs using `INTER_NR_test2.tbl` from sample ISIS data.
- Ensure that processing runs creates a binned MomentumTransfer workspace "_IvsQ_binned" and an unbinned workspace "_IvsQ".
  - Ensure that it works for group workspaces (in which unbinned workspaces should also be grouped).
  - Go to Options->Refl Gui Options and uncheck "Use ReflectometryReductionOneAuto" Algorithm and check that processing runs produces the same result.
- Login to the catalog and enter `1520249` into 'RB Search' - the left column should be populated with runs. Select them all and click the transfer button - they should successfully transfer.
- Go to File->Save Workspaces - ensure that the save path is filled correctly (not missing /).

Fixes #18185.

*Release notes have been updated.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?

- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by raquelalvarezbanos** [comment]
I noticed that when transferring runs, the angle is not transferred properly, `dq/Q` is also transferred to the column, which causes the reduction to fail.

**Comment by raquelalvarezbanos** [comment]
Something must be wrong: reducing runs 13460 and 13462 (INTER), the results are:

gui

`13460_Ivs_Q_binned` should be just `13460_IvsQ` with a rebin step, not a shift in X.

However, scientists requested last week that the rebinned workspace should be an additional output of `ReflectometryReductionOneAuto`, this will be fixed in PR https://github.com/mantidproject/mantid/pull/18058, so let's wait for that PR to be merged in and use those changes in this PR.

**Comment by PranavBahuguna** [comment]
I think this is because I obtained the unbinned workspace by simply running ConvertUnits on the final workspace, I believe there is more it than simply doing that. Regardless, it wont be necessary anyway once PR #18058 is merged.

**Comment by martyngigg** [comment]
@PranavBahuguna Some conflicts have arisen.

File: scripts/Interface/ui/reflectometer/refl_gui.py [hunk 1]
```diff
@@ -1028,24 +1038,44 @@ def _do_run(self, runno, row, which):
...
+                                                               ScaleFactor=factor,MomentumTransferStep=Qstep,
+                                                               MomentumTransferMinimum=Qmin, MomentumTransferMaximum=Qmax,
+                                                               Version=1)
+
+                wqBinned, wq, wlam = ReflectometryReductionOneAuto(InputWorkspace=ws, FirstTransmissionRun=transmission_ws,
+                                                                   thetaIn=angle, OutputWorkspaceBinned=runno+'_IvsQ_binned',
+                                                                   OutputWorkspace=runno+'_IvsQ',
+                                                                   OutputWorkspaceWavelength=runno+'_IvsLam',
+                                                                   ScaleFactor=factor,MomentumTransferStep=Qstep,
+                                                                   MomentumTransferMin=Qmin, MomentumTransferMax=Qmax)
```
Comment (raquelalvarezbanos) [inline scripts/Interface/ui/reflectometer/refl_gui.py]: I think I came up with a better solution: If you replace lines `1066` to `1078` with:

```
                alg = AlgorithmManager.create("ReflectometryReductionOneAuto")
                alg.initialize()
                alg.setProperty("InputWorkspace", ws)
                alg.setProperty("FirstTransmissionRun", transmission_ws)
                if (angle != None):
                    alg.setProperty("ThetaIn", angle)
                alg.setProperty("OutputWorkspaceBinned", runno+'_IvsQ_binned')
                alg.setProperty("OutputWorkspace", runno+'_IvsQ')
                alg.setProperty("OutputWorkspaceWavelength", runno+'_IvsLam')
                alg.setProperty("ScaleFactor", factor)
                alg.setProperty("MomentumTransferStep", Qstep)
                alg.setProperty("MomentumTransferMin", Qmin)
                alg.setProperty("MomentumTransferMax", Qmax)
                alg.execute()
                wlam = mtd[runno+'_IvsLam']
                wq = mtd[runno+'_IvsQ']
                wqBinned = mtd[runno+'_IvsQ_binned']
                th = alg.getProperty("ThetaIn").value
```

we get the same functionality without having to run the algorithm twice. The same should work for lines `1042` - `1055`, but I haven't tried yet.

**Comment by raquelalvarezbanos** [comment]
- Runs are transferred correctly (run number, angle and dq/q).
- Forward slash was added to the save path.
- Processing creates two `IvsQ` workspace, one with native binning and one that has been rebinned. Both are compatible.
- Columns `Angle`, `qmin`, `qmax` and `dq/q` are automatically populated when not given.

:shipit:

**Comment by martyngigg** [comment]
@PranavBahuguna You appear to have deleted `docs/source/images/release38.png`, which is the release image for 3.8. Can you put this back, please? It's good to then.

**Comment by martyngigg** [comment]
:+1:

**Comment by AndreiSavici** [comment]
change the base branch to release-v3.9

**Comment by martyngigg** [comment]
@PranavBahuguna Can you edit the base branch to `release-v3.9`, please?

**Pull request merged**